File: CustomerApp/app/src/main/java/com/example/myapplication/ui/fragment/home/SectionsPagerAdapter.java
package com.example.myapplication.ui.fragment.home;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.example.myapplication.ui.fragment.menu.today.TodaysMenuFragment;
import com.example.myapplication.ui.fragment.menu.tomorrow.TomorrowMenuFragment;
import com.example.myapplication.ui.fragment.menu.restoftheweek.RestoftheWeekFragment;

class SectionsPagerAdapter extends FragmentPagerAdapter {
    SectionsPagerAdapter(FragmentManager fm) {
        super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return TodaysMenuFragment.newInstance();
            case 1:
                return TomorrowMenuFragment.newInstance();
            case 2:
                return RestoftheWeekFragment.newInstance();
            default:
                throw new IllegalArgumentException("Position " + position + " is out of bounds!");
        }
    }

    @Override
    public int getCount() {
        return 3;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return "Today's Menu";
            case 1:
                return "Tomorrow's Menu ";
            case 2:
                return "Rest of the Week ";
            default:
                break;
        }
        return null;
    }
}

File: CustomerApp/app/src/main/java/com/example/myapplication/ui/fragment/home/SectionsPagerAdapter.kt
package com.example.myapplication.ui.fragment.home

import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.example.myapplication.ui.fragment.menu.restoftheweek.RestOfTheWeekMenuFragment
import com.example.myapplication.ui.fragment.menu.today.TodayMenuFragment
import com.example.myapplication.ui.fragment.menu.tomorrow.TomorrowMenuFragment
import kotlinx.coroutines.ExperimentalCoroutinesApi

@ExperimentalCoroutinesApi
internal class SectionsPagerAdapter(fragment: Fragment): FragmentStateAdapter(fragment) {
    override fun createFragment(position: Int): Fragment = when (position) {
        0 -> TodayMenuFragment.newInstance()
        1 -> TomorrowMenuFragment.newInstance()
        2 -> RestOfTheWeekMenuFragment.newInstance()
        else -> throw IllegalArgumentException("Position $position is out of bounds!")
    }

    override fun getItemCount(): Int = 3
}